run.py

```diff
@@ -1,16 +1,42 @@
+from flask import Flask, request
+import jwt
+import requests
 import os
 
-from flask import Flask
-from jinja2 import Template
-
 app = Flask(__name__)
 
-t = Template("Hello World!")
+IAP_AUDIENCE = os.getenv('IAP_AUDIENCE', None)
+
+
+def get_jwt():
+    if IAP_AUDIENCE:
+        try:
+            iap_jwt = request.headers['x-goog-iap-jwt-assertion']
+            key = get_iap_public_key(jwt.get_unverified_header(iap_jwt).get('kid'))
+            decoded_jwt = jwt.decode(iap_jwt, key, algorithms=['ES256'], audience=IAP_AUDIENCE)
+            return decoded_jwt
+        except KeyError:
+            pass
+
+    return {"email": "bananas@noodle.poodle"}
+
+
+def get_iap_public_key(key_id):
+    if key_id not in get_iap_public_key.cache:
+        resp = requests.get('https://www.gstatic.com/iap/verify/public_key')
+        resp.raise_for_status()
+        get_iap_public_key.cache = resp.json()
+
+    return get_iap_public_key.cache[key_id]
+
+
+get_iap_public_key.cache = {}
```

Comment:
`functools.lru_cache` is a nice built in way of caching function results like this

Reply:
I don't know how to do that. Might have to pair on it to take advantage of your Pyhthonista skillz